### DISCUSSION TITLE
After Container Crashed

### DISCUSSION THREAD

**Discussion by zhoub** [k3s-io/k3s#4206] [General]
Hi,

Our Pod(program) in container might need a lot of memory to process data, and during our tests once the Pod(program) requested too much memory that exceeded the Node(physical machine) could offer, the k3s cluster enters a strange mode:

1. Any operation to the problematic Development didn't have any response
2. Command `kubectl get nodes` returned `no metric server`
3. Couldn't delete any ReplicaSet

The only workable solution is to restart k3s server, then k3s would
1. Clean up all legacy ReplicaSet with Pod 0/0
2. Problematic Development could restart

And after a while, everything would be back to normal.

Using
```
k3s version v1.22.2+k3s1 (10bca343)
go version go1.16.8
```

Is this a bug at k3s which could be fixed or improved ?

Thanks in advance !

**Comment by dweomer** [comment]
> Is this a bug at k3s which could be fixed or improved ?

Nope. It's a bug in your deployment. Consider placing resource requests+limits on your containers to prevent them from making your node unresponsive. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container
